llvm/include/llvm/CodeGen/RegAllocCommon.h

```diff
@@ -13,19 +13,21 @@
 
 namespace llvm {
 
-class TargetRegisterClass;
 class TargetRegisterInfo;
+class MachineRegisterInfo;
+class Register;
 
 typedef std::function<bool(const TargetRegisterInfo &TRI,
-                           const TargetRegisterClass &RC)> RegClassFilterFunc;
+                           const MachineRegisterInfo &MRI, const Register &Reg)>
+    RegClassFilterFunc;
 
 /// Default register class filter function for register allocation. All virtual
 /// registers should be allocated.
 static inline bool allocateAllRegClasses(const TargetRegisterInfo &,
-                                         const TargetRegisterClass &) {
+                                         const MachineRegisterInfo &,
+                                         const Register &) {
```

Comment:
Register should be passed by value

Reply:
will do

Reply:
I tried it, pass-by-value does not work in this case. With MSVC, I am getting the following error:

error C2027: use of undefined type 'llvm::Register'

It looks like this header file does not want any dependence on other header file

Reply:
adding llvm/CodeGen/Register.h seems okay. fixed the error